Firmware/ultralcd.cpp

```diff
@@ -5365,6 +5365,9 @@ static void lcd_main_menu()
                             MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
                         }
                     } else {
+                        if (!fsensor.getAutoLoadEnabled()) {
+                            MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
+                        }
```

Comment:
I think this logic can be simplified to this:

```cpp
if (fsensor.isEnabled()) {

    if (!fsensor.getAutoLoadEnabled()) {
        MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
    }

    if (!fsensor.getFilamentPresent()) {
        if (fsensor.getAutoLoadEnabled()) {
            MENU_ITEM_SUBMENU_P(_T(MSG_AUTOLOAD_FILAMENT), lcd_menu_AutoLoadFilament);
        }
    } else {
        MENU_ITEM_SUBMENU_P(_T(MSG_UNLOAD_FILAMENT), lcd_unLoadFilament);
    }
}
```

Perhaps we can remove `lcd_menu_AutoLoadFilament`? That menu item doesn't really do anything. Then the logic could look like this:

```cpp
if (fsensor.isEnabled()) {

    if (!fsensor.getAutoLoadEnabled()) {
        MENU_ITEM_SUBMENU_P(_T(MSG_LOAD_FILAMENT), lcd_LoadFilament);
    }

    if (fsensor.getFilamentPresent()) {
        MENU_ITEM_SUBMENU_P(_T(MSG_UNLOAD_FILAMENT), lcd_unLoadFilament);
    }
}
```

Reply:
@3d-gussner Is it possible for us to remove `lcd_menu_AutoLoadFilament`? I feel it should suffice for the user to open the filament settings to view if autoloading is enabled or not.

Reply:
I like the idea of removing it.
I added the removal as a separate commit, so you can choose if you want to merge it.